### PR TITLE
I63 uk dataset collation

### LINKED ISSUE #63: csv output from new UK data sources
Linked to #62 , adding new data sources for UK estimates.

For these to be used, ideally the summary of Rt estimates needs to be output in a single csv file. 

This should
 - Combine Rt estimates from each data source 
 - Include a data source identifier
 - Filter the `type` variable to only the Rt "estimates" (not the estimates from partial data or forecasts)
 - Save in a csv in a dedicated folder within the `united-kingdom` folder.

Below is example code to get to what is needed. In this example the new dedicated folder for output is called "all-summary-rt".

However I am not sure where this needs to go to get it running in the process after the daily Rt estimates have finished. 

```
# Get Rt estimates summary from each data source
cases <- fread(here::here("subnational", "united-kingdom", "cases", "summary", "rt.csv"))
cases <- cases[cases$type == "estimate"][, data_source := "test-positive cases"]

deaths <- fread(here::here("subnational", "united-kingdom", "deaths", "summary", "rt.csv"))
deaths <- deaths[deaths$type == "estimate"][, data_source := "deaths"]

admissions <- fread(here::here("subnational", "united-kingdom", "admissions", "summary", "rt.csv"))
admissions <- admissions[admissions$type == "estimate"][, data_source := "hospital admissions"]

# Bind all data sources
uk_rt <- data.table::rbindlist(list(cases, deaths, admissions),
                               fill = TRUE, use.names = TRUE)

# Save back to main UK folder
write.csv(uk_rt, here::here("subnational", "united-kingdom", "all-summary-rt", paste0(Sys.Date(), "-uk-rt.csv"))
```

### PR DESCRIPTION
**Pull request by joeHickson**
closes #63

### CODE REVIEW

**Comment by joeHickson** [comment]
@kathsherratt / @seabbs  can you check the output file is as you expect and the prefix is acceptable?

File: R/utils.R [hunk 1]
```diff
@@ -152,11 +152,11 @@ collate_estimates <- function(name, target = "rt"){
   
   # Check a collated file exists
   if(!dir.exists(here::here("subnational", name, "collated", target))){
-    dir.create(here::here("subnational", name, "collated", target))
+    dir.create(here::here("subnational", name, "collated", target), recursive = TRUE)
   }
   
   # Save back to main UK folder
-  data.table::fwrite(df, here::here("subnational", name, "collated", target, paste0(Sys.Date(), ".csv")))
+  data.table::fwrite(df, here::here("subnational", name, "collated", target, paste0('summary_',Sys.Date(), ".csv")))
```
Comment (seabbs) [inline R/utils.R:159]: not sure this is needed as the expectation is that the collated estimates are summaries?

File: R/run-region-updates.R [hunk 2]
```diff
@@ -200,16 +205,16 @@ if (sys.nframe() == 0) {
...
-    #                              futile.logger::flog.warn(w)
-    #                              rlang::cnd_muffle(w)
-    #                            },
-    #                            error = function(e) {
-    #                              futile.logger::flog.error(capture.output(rlang::trace_back()))
-    #                            }),
-    #        error = function(e) {
-    #          futile.logger::flog.error(e)
-    #        })
+  run_regional_updates(datasets = datasets, args = args)
```
Comment (seabbs) [inline R/run-region-updates.R:208]: Not for this PR but this hack job was from a quick fix from me whilst you were on holiday and means we throw away errors (oops).

File: R/utils.R (hunk 1 above)
Comment (joeHickson) [reply R/utils.R:159]: without it the gitignore file hits it as being in the format [0-9]* so ignores the file

File: R/run-region-updates.R (hunk 2 above)
Comment (joeHickson) [reply R/run-region-updates.R:208]: I had spotted... I'm sure there was a reason but we should reintroduce it. Perhaps the ftry method would be better?

Comment (seabbs) [reply R/run-region-updates.R:208]: it was the issue of an object being passed causing a silent immediate crash - a slight nightmare to debug! Yes `ftry` approach seems sensible.

File: R/utils.R (hunk 1 above)
Comment (seabbs) [reply R/utils.R:159]: ah - good spot! All good then.

**Comment by seabbs** [comment]
Output file matches my expectations. @kathsherratt ?

File: R/utils.R (hunk 1 above)
Comment (joeHickson) [reply R/utils.R:159]: my test run was refusing to show any new files to commit and when I went digging it was there...

File: R/run-region-updates.R (hunk 2 above)
Comment (joeHickson) [reply R/run-region-updates.R:208]: raised #70  so it doesn't get forgotten

**Comment by kathsherratt** [comment]
Output looks ideal, thanks!

**Pull request merged**